repair/repair.hh

```diff
@@ -183,6 +184,7 @@ public:
     uint64_t _sub_ranges_nr = 0;
 public:
     repair_info(seastar::sharded<database>& db_,
+            const dht::i_partitioner& p,
             const sstring& keyspace_,
```

Comment:
Please add a comment here that the partitioner must match the tables. Maybe it's better to move the check here, it's cleaner not to depend on the caller.

Reply:
In https://github.com/scylladb/scylla/pull/4975, we use repair_info in sync_data_using_repair. If we move the check into repair_info, we can avoid duplicating the check code. So please do so.

Reply:
I moved the check to repair_info constructor.